lib/apphook.c

```diff
@@ -215,4 +216,6 @@ app_thread_stop(void)
   main_loop_call_thread_deinit();
   dns_caching_thread_deinit();
   scratch_buffers_allocator_deinit();
+  g_atomic_int_dec_and_test(&main_loop_workers_running);
+  g_cond_signal(&thread_halt_cond);
 }
```

Comment:
I think this should probably go the call-sites where app_thread_start/stop is getting called.